test/functional/legacy/025_jump_tag_hidden_spec.lua

```diff
@@ -6,7 +6,14 @@ local clear, feed, insert = helpers.clear, helpers.feed, helpers.insert
 local feed_command, expect = helpers.feed_command, helpers.expect
 
 describe('jump to a tag with hidden set', function()
-  setup(clear)
+  setup(function ()
+    clear()
+    helpers.rmdir('test25.dir')
+  end)
+
+  teardown(function ()
+    helpers.rmdir('test25.dir')
```

Comment:
@justinmk This deleted the current directory even though `test25.dir` is suppose to be a symlink. I assume other tests don't use `test25.dir`.

Reply:
What do you mean "even though test25.dir is suppose to be a symlink"? Do you expect it only to delete the symlink, but it's deleting the target directory?

That's because `helpers.do_rmdir()` recurses into the directory contents, and doesn't check if it's a symlink. Probably `helpers.rmdir()` is not the right thing to use if one only wants to delete a symlink.

Reply:
Why doesn't `helpers.do_rmdir` check if it's a symlink?

Reply:
Just no one added that feature. And the main purpose was to recursively clear directories.

In the tests we always know if something we created is a symlink, right? So there is not much reason to use rmdir in that case.

Reply:
If `helpers.rmdir` doesn't check if it's a symlink or not, then the tester must explicity check the file attributes themselves especially if they're "sharing" (same filename) files with other tests. This means we can't reliably use it for legacy tests. There's already a Travis build failure on legacy test 031 after using the modified `helpers.rmdir` to handle symlinks on legacy test 025.

Were there issues with `rm -rf` or using `delete()` directly instead of LuaFileSystem?

Reply:
https://github.com/neovim/neovim/pull/4936 is the problem in Windows only? I don't have issues with cmd.exe's `rd`.